test/e2e/functional/tests/e2e.go

```diff
@@ -354,6 +365,826 @@ var _ = Describe("Ingress Node Firewall", func() {
+			additionalServerPodName              = "e2e-inf-rules-additional-server-"
+			additionalAdditionalServerPodName    = "e2e-inf-rules-additional-additional-server-"
+			additionalAdditionalServerListenPort = "8080"
+			defaultSourcePort                    = "40000"
+			clientPodName                        = "e2e-inf-rules-client-"
+			podLabel                             = map[string]string{"e2e-inf-rules": ""}
+		)
+
+		BeforeEach(func() {
+			//setup INF config
+			config = &ingressnodefwv1alpha1.IngressNodeFirewallConfig{}
+			err := infwutils.LoadIngressNodeFirewallConfigFromFile(config, consts.IngressNodeFirewallConfigCRFile)
+			Expect(err).ShouldNot(HaveOccurred())
+			config.SetNamespace(OperatorNameSpace)
+			err = infwutils.EnsureIngressNodeFirewallConfigExists(testclient.Client, config, timeout)
+			Expect(err).ShouldNot(HaveOccurred())
+			//wait for daemonset to be rolled out
+			infDaemonSet := &appsv1.DaemonSet{}
+			err = daemonset.WaitForDaemonSetReady(testclient.Client, infDaemonSet, OperatorNameSpace,
+				"ingress-node-firewall-daemon", retryInterval, timeout)
+			Expect(err).ShouldNot(HaveOccurred())
+			//setup client pod which we always need
+			client = pods.GetNetcatClientPod(clientPodName, OperatorNameSpace, podLabel)
+			client, err = pods.EnsurePodRunning(testclient.Client, client, OperatorNameSpace, retryInterval, timeout)
+			Expect(err).ShouldNot(HaveOccurred())
+			server = pods.GetNetcatServerPod(serverPodName, OperatorNameSpace, serverListenPort, podLabel,
+				ingressnodefwv1alpha1.ProtocolTypeTCP)
+			server, err = pods.EnsurePodRunning(testclient.Client, server, OperatorNameSpace, retryInterval, timeout)
+			Expect(err).Should(BeNil())
+			inf = &ingressnodefwv1alpha1.IngressNodeFirewall{}
+			inf.SetName("e2e-inf-tcp-rules")
+			infwutils.DefineWithWorkerNodeSelector(inf)
+			infwutils.DefineWithInterface(inf, testInterface)
+			infwutils.AppendIngress(inf, fmt.Sprintf("%s/32", client.Status.PodIP),
+				infwutils.GetTCPRule(1, serverListenPort, ingressnodefwv1alpha1.IngressNodeFirewallDeny))
+			Expect(infwutils.CreateIngressNodeFirewall(testclient.Client, inf, timeout)).ShouldNot(HaveOccurred())
+			// TODO(mk): remove this sleep when we have a way to detect INF application is completed. Currently, we have no
+			// transition or pending state in the INF obj.
+			time.Sleep(200 * time.Millisecond)
+		})
+
+		AfterEach(func() {
+			pods.DeletePod(testclient.Client, client, timeout)
+			pods.DeletePod(testclient.Client, server, timeout)
+			infwutils.DeleteIngressNodeFirewall(testclient.Client, inf, timeout)
+			infwutils.DeleteINFConfig(config)
+		})
+
+		It("block a single port", func() {
+			Eventually(func() bool {
+				_, stdErr, err := pods.NetcatTCPV4(testclient.Client, client, server.Status.PodIP, defaultSourcePort,
+					serverListenPort, timeout)
+				if stdErr != "" || err != nil {
+					return true
+				}
+				return false
+			}, timeout, retryInterval)
```

Comment:
will be sweet to check when the block happens event is fired up for the same protocol and port

Reply:
you mean how long it takes (latency) for it to block?

Reply:
Dont think I understand the ask.